### DISCUSSION TITLE
[Support]: Frigate retention - cleaning up when plenty of storage is left - unstable camera connection?

### DISCUSSION THREAD

**Discussion by altimeter** [blakeblackshear/frigate#13666] [General Support]
### Describe the problem you are having

Hi there,

Thank you so much for your amazing work on Frigate. I’m experiencing a retention issue after the introduction of the retention policy in version 0.14, which automatically cleans up recordings before storage runs out.

I’ve noticed that this clean-up process seems to be triggered incorrectly, possibly due to an unstable Wi-Fi connection with my cameras, particularly the camera. The logs indicate a connection issue with this camera.

Whenever I enable recording, after a short time, Frigate starts reporting that it's running out of storage and begins cleaning up recordings for all cameras.

It uses a variable bitrate with a maximum of 2Mb/s, but it never reaches that limit. The total recording size is only around 1GB for the entire day, while the total size for all my cameras is around 25-30GB. My disk still has 141GB of free space.

Could you please review below details and give any advice? Also, feel free to comment on my config file for any other issues.

Thanks again!

docker exec -it frigate sh
df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay         234G   81G  141G  37% /
tmpfs            64M     0   64M   0% /dev
shm             200M   14M  187M   7% /dev/shm
/dev/sdb1       916G  593G  277G  69% /config
tmpfs           954M   86M  869M   9% /tmp/cache
/dev/sda2       234G   81G  141G  37% /usr/share/zoneinfo/Etc/UTC

### Version

0.14.1-f4f3cfa

**Comment by NickM-27** [comment]
> introduction of the retention policy in version 0.14, which automatically cleans up recordings before storage runs out.

This has been in frigate since 0.12

Your best bet is to enable debug logs for frigate.storage

**Reply by altimeter** [reply]
Frigate's storage system was incorrectly displaying the bandwidth for the camera as 166998.77 MiB/hr, which was triggering unnecessary recording cleanups. However, the actual bitrate for the camera was only around 350kb/s at the time. Any suggestions on how to resolve this issue? Thanks!

**Reply by NickM-27** [reply]
The bandwidth calculation is very simple, it just takes the amount of time each of the segments takes up along with the size of the file to calculate bandwidth. Important to remember camera bandwidth doesn't tell the whole story. I'd suggest looking into the file system and see what the file sizes and durations are for this particular camera

**Reply by altimeter** [reply]
The problem is that whenever the auto-cleaning process is mistakenly triggered, the recordings are deleted, making it impossible to review them. However, I had a separate ffmpeg process recording from the camera, and the file size was quite small, indicating the bitrate was only around 350kb/s at that time. 

Is there a way to disable the auto-cleaning? Or, is there an alternative retention mode, such as only keeping recordings with active objects, that would avoid triggering the auto-cleaning process?

**Reply by NickM-27** [reply]
What command did you run to create the recordings? It's not always the same. 

And no it's not possible to disable the emergency cleanup function

**Reply by altimeter** [reply]
I just simply copy the rtsp stream via ffmpeg command: "ffmpeg -hide_banner -y -loglevel error -rtsp_transport udp -use_wallclock_as_timestamps 1 -i rtsp://url -vcodec copy -acodec copy -f segment -reset_timestamps 1 -segment_time 300 -segment_format mkv -segment_atclocktime 1 -strftime 1 '/camera.mkv'"

**Reply by NickM-27** [reply]
yeah, that's different than what frigate does, for example the segment times are much shorter which we have seen causes issues with some buggy cameras as their iframe timing is inconsistent. You are also using mkv but frigate uses mp4. Best to use the `/api/stats` to find the ffmpeg command for the particular camera and use it directly.

**Reply by altimeter** [reply]
Update: I’ve found that any unstable Wi-Fi camera connection, not just CAR, can trigger the issue. I’ll attempt to run a parallel command similar to Frigate’s, as shown below, to record.

ffmpeg -hide_banner -loglevel warning -threads 2 -hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format vaapi -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport udp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://xxxx -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac "/xxxx/cam@%Y%m%d%H%M%S%z.mp4" -r 10 -vf fps=10,scale_vaapi=w=1920:h=1080:format=nv12,hwdownload,format=nv12,format=yuv420p -threads 2 -f rawvideo -pix_fmt yuv420p
